Switching Vert.x app to RestEasy Reactive, fails debugging session

I have a patch that ports the Vert.x test app to contemporary Quarkus 3.x suggested API, RestEasy Reactive.

[quarkus_3.x.reactive.patch.txt](https://github.com/Karm/mandrel-integration-tests/files/13486026/quarkus_3.x.reactive.patch.txt)

It fails debugging session:

"Cannot execute this command while the selected thread is running"

```
[ERROR] Failures: 
[ERROR]   DebugSymbolsTest.debugSymbolsQuarkusContainer:396->carryOutGDBSession:501 There were errors in the GDB session. Note that commands in the session might depend on each other. Errors: 
Command 'bt' did not match the expected pattern in time '.*at org/acme/vertx/Fruit.java:48.*'.
Output was:
"bt
""Selected thread is running.
"^error,msg="Selected thread is running."(gdb) , 
Command 'list' did not match the expected pattern in time '.*48.*return client.query\("SELECT id, name FROM fruits ORDER BY name ASC.*'.
Output was:
"list
""1     org/graalvm/compiler/replacements/BoxingSnippets.java: No such file or directory.
"^done(gdb) , 
Command 'c&' did not match the expected pattern in time '.*Continuing.*'.
Output was:
"c&
""Cannot execute this command while the selected thread is running.
"^error,msg="Cannot execute this command while the selected thread is running."(gdb)  ==> expected: <true> but was: <false>
[INFO] 
[ERROR] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0
```

--- a/.github/workflows/builder_image_tests.yml
+++ b/.github/workflows/builder_image_tests.yml
@@ -27,14 +27,16 @@ jobs:
       matrix:
         os: [ ubuntu-20.04 ]
         include:
-          - quarkus-version: '2.13.8.Final'
+          - quarkus-version: '2.13.9.Final'
             mandrel-builder-image: '22.3-java17'
-          - quarkus-version: '2.16.11.Final'
+          - quarkus-version: '2.16.12.Final'
             mandrel-builder-image: '22.3-java17'
-          - quarkus-version: '3.2.6.Final'
+          - quarkus-version: '3.2.9.Final'
             mandrel-builder-image: 'jdk-17'
-          - quarkus-version: '3.2.6.Final'
-            mandrel-builder-image: 'jdk-20'
+          - quarkus-version: '3.5.3'
+            mandrel-builder-image: 'jdk-21'
+          - quarkus-version: '3.6.0'
+            mandrel-builder-image: 'jdk-21'
     steps:
       - uses: actions/checkout@v3
         with:

--- a/.github/workflows/local_tests.yml
+++ b/.github/workflows/local_tests.yml
@@ -26,46 +26,58 @@ jobs:
       fail-fast: false
       matrix:
         include:
-          - quarkus-version: '2.13.8.Final'
-            mandrel-version: '22.3.3.1-Final'
+          - quarkus-version: '2.13.9.Final'
+            mandrel-version: '22.3.4.0-Final'
             java-version: '17'
             os: 'windows-2019'
             timeout-minutes: 80
-          - quarkus-version: '2.13.8.Final'
-            mandrel-version: '22.3.3.1-Final'
+          - quarkus-version: '2.13.9.Final'
+            mandrel-version: '22.3.4.0-Final'
             java-version: '17'
             os: 'ubuntu-20.04'
             timeout-minutes: 80
 
-          - quarkus-version: '2.16.11.Final'
-            mandrel-version: '22.3.3.1-Final'
+          - quarkus-version: '2.16.12.Final'
+            mandrel-version: '22.3.4.0-Final'
             java-version: '17'
             os: 'windows-2019'
             timeout-minutes: 80
-          - quarkus-version: '2.16.11.Final'
-            mandrel-version: '22.3.3.1-Final'
+          - quarkus-version: '2.16.12.Final'
+            mandrel-version: '22.3.4.0-Final'
             java-version: '17'
             os: 'ubuntu-20.04'
             timeout-minutes: 80
 
-          - quarkus-version: '3.2.6.Final'
-            mandrel-version: '23.0.1.2-Final'
-            java-version: '20'
+          - quarkus-version: '3.2.9.Final'
+            mandrel-version: '23.0.2.1-Final'
+            java-version: '17'
             os: 'windows-2022'
             timeout-minutes: 80
-          - quarkus-version: '3.2.6.Final'
-            mandrel-version: '23.0.1.2-Final'
-            java-version: '20'
+          - quarkus-version: '3.2.9.Final'
+            mandrel-version: '23.0.2.1-Final'
+            java-version: '17'
             os: 'ubuntu-20.04'
             timeout-minutes: 130
-          - quarkus-version: '3.2.6.Final'
-            mandrel-version: '23.0.1.2-Final'
-            java-version: '17'
+
+          - quarkus-version: '3.5.3'
+            mandrel-version: '23.1.1.0-Final'
+            java-version: '21'
             os: 'windows-2022'
             timeout-minutes: 80
-          - quarkus-version: '3.2.6.Final'
-            mandrel-version: '23.0.1.2-Final'
-            java-version: '17'
+          - quarkus-version: '3.5.3'
+            mandrel-version: '23.1.1.0-Final'
+            java-version: '21'
+            os: 'ubuntu-20.04'
+            timeout-minutes: 130
+
+          - quarkus-version: '3.6.0'
+            mandrel-version: '23.1.1.0-Final'
+            java-version: '21'
+            os: 'windows-2022'
+            timeout-minutes: 80
+          - quarkus-version: '3.6.0'
+            mandrel-version: '23.1.1.0-Final'
+            java-version: '21'
             os: 'ubuntu-20.04'
             timeout-minutes: 130
     steps:

--- a/.github/workflows/testing_testsuite.yml
+++ b/.github/workflows/testing_testsuite.yml
@@ -22,7 +22,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        mandrel-version: [ '22.3.3.1-Final', '23.0.1.2-Final' ]
+        mandrel-version: [ '22.3.4.0-Final', '23.0.2.1-Final' ]
         java-version: [ '17' ]
         os: [ ubuntu-20.04, windows-2019 ]
     steps:

--- a/apps/quarkus-vertx/quarkus_3.x.patch
+++ b/apps/quarkus-vertx/quarkus_3.x.patch
@@ -1,8 +1,38 @@
+diff --git a/apps/quarkus-vertx/pom.xml b/apps/quarkus-vertx/pom.xml
+index 7d47c9b..b52d46c 100644
+--- a/apps/quarkus-vertx/pom.xml
++++ b/apps/quarkus-vertx/pom.xml
+@@ -36,11 +36,7 @@
+     <dependencies>
+         <dependency>
+             <groupId>io.quarkus</groupId>
+-            <artifactId>quarkus-resteasy</artifactId>
+-        </dependency>
+-        <dependency>
+-            <groupId>io.quarkus</groupId>
+-            <artifactId>quarkus-resteasy-mutiny</artifactId>
++            <artifactId>quarkus-resteasy-reactive</artifactId>
+         </dependency>
+         <dependency>
+             <groupId>io.quarkus</groupId>
+@@ -56,11 +52,7 @@
+         </dependency>
+         <dependency>
+             <groupId>io.quarkus</groupId>
+-            <artifactId>quarkus-resteasy-jsonb</artifactId>
+-        </dependency>
+-        <dependency>
+-            <groupId>io.smallrye.reactive</groupId>
+-            <artifactId>smallrye-mutiny-vertx-web-client</artifactId>
++            <artifactId>quarkus-resteasy-reactive-jsonb</artifactId>
+         </dependency>
+     </dependencies>
+     <build>
 diff --git a/apps/quarkus-vertx/src/main/java/org/acme/vertx/FruitResource.java b/apps/quarkus-vertx/src/main/java/org/acme/vertx/FruitResource.java
-index cf0796c..66c53c2 100644
+index cf0796c..8cb6b0c 100644
 --- a/apps/quarkus-vertx/src/main/java/org/acme/vertx/FruitResource.java
 +++ b/apps/quarkus-vertx/src/main/java/org/acme/vertx/FruitResource.java
-@@ -18,19 +18,19 @@ package org.acme.vertx;
+@@ -18,22 +18,22 @@ package org.acme.vertx;
  
  import java.net.URI;
  
@@ -27,6 +57,7 @@ index cf0796c..66c53c2 100644
 +import jakarta.ws.rs.POST;
 +import jakarta.ws.rs.PUT;
 +import jakarta.ws.rs.Path;
++import jakarta.ws.rs.PathParam;
 +import jakarta.ws.rs.Produces;
 +import jakarta.ws.rs.core.MediaType;
 +import jakarta.ws.rs.core.Response;
@@ -34,28 +65,109 @@ index cf0796c..66c53c2 100644
 +import jakarta.ws.rs.core.Response.Status;
  
  import org.eclipse.microprofile.config.inject.ConfigProperty;
- import org.jboss.resteasy.annotations.jaxrs.PathParam;
+-import org.jboss.resteasy.annotations.jaxrs.PathParam;
+ 
+ import io.smallrye.mutiny.Uni;
+ import io.vertx.mutiny.pgclient.PgPool;
+@@ -43,29 +43,9 @@ import io.vertx.mutiny.pgclient.PgPool;
+ @Consumes(MediaType.APPLICATION_JSON)
+ public class FruitResource {
+ 
+-    @Inject
+-    @ConfigProperty(name = "myapp.schema.create", defaultValue = "true")
+-    boolean schemaCreate;
+-
+     @Inject
+     PgPool client;
+ 
+-    @PostConstruct
+-    void config() {
+-        if (schemaCreate) {
+-            initdb();
+-        }
+-    }
+-
+-    private void initdb() {
+-        client.query("DROP TABLE IF EXISTS fruits").execute()
+-                .flatMap(r -> client.query("CREATE TABLE fruits (id SERIAL PRIMARY KEY, name TEXT NOT NULL)").execute())
+-                .flatMap(r -> client.query("INSERT INTO fruits (name) VALUES ('Orange')").execute())
+-                .flatMap(r -> client.query("INSERT INTO fruits (name) VALUES ('Pear')").execute())
+-                .flatMap(r -> client.query("INSERT INTO fruits (name) VALUES ('Apple')").execute())
+-                .await().indefinitely();
+-    }
+-
+     @GET
+     public Uni<Response> get() {
+         return Fruit.findAll(client)
+@@ -75,7 +55,7 @@ public class FruitResource {
+ 
+     @GET
+     @Path("{id}")
+-    public Uni<Response> getSingle(@PathParam Long id) {
++    public Uni<Response> getSingle(@PathParam(value = "id") Long id) {
+         return Fruit.findById(client, id)
+                 .onItem().transform(fruit -> fruit != null ? Response.ok(fruit) : Response.status(Status.NOT_FOUND))
+                 .onItem().transform(ResponseBuilder::build);
+@@ -90,7 +70,7 @@ public class FruitResource {
+ 
+     @PUT
+     @Path("{id}")
+-    public Uni<Response> update(@PathParam Long id, Fruit fruit) {
++    public Uni<Response> update(@PathParam(value = "id") Long id, Fruit fruit) {
+         return fruit.update(client)
+                 .onItem().transform(updated -> updated ? Status.OK : Status.NOT_FOUND)
+                 .onItem().transform(status -> Response.status(status).build());
+@@ -98,7 +78,7 @@ public class FruitResource {
+ 
+     @DELETE
+     @Path("{id}")
+-    public Uni<Response> delete(@PathParam Long id) {
++    public Uni<Response> delete(@PathParam(value = "id") Long id) {
+         return Fruit.delete(client, id)
+                 .onItem().transform(deleted -> deleted ? Status.NO_CONTENT : Status.NOT_FOUND)
+                 .onItem().transform(status -> Response.status(status).build());
 diff --git a/apps/quarkus-vertx/src/main/java/org/acme/vertx/GreetingResource.java b/apps/quarkus-vertx/src/main/java/org/acme/vertx/GreetingResource.java
-index ac167a0..c5c66fa 100644
+index ac167a0..1b56c15 100644
 --- a/apps/quarkus-vertx/src/main/java/org/acme/vertx/GreetingResource.java
 +++ b/apps/quarkus-vertx/src/main/java/org/acme/vertx/GreetingResource.java
-@@ -18,11 +18,11 @@ package org.acme.vertx;
- import static java.util.concurrent.TimeUnit.MILLISECONDS;
- import static java.util.concurrent.TimeUnit.NANOSECONDS;
+@@ -15,19 +15,17 @@
+  */
+ package org.acme.vertx;
  
+-import static java.util.concurrent.TimeUnit.MILLISECONDS;
+-import static java.util.concurrent.TimeUnit.NANOSECONDS;
+-
 -import javax.inject.Inject;
 -import javax.ws.rs.GET;
 -import javax.ws.rs.Path;
 -import javax.ws.rs.Produces;
 -import javax.ws.rs.core.MediaType;
+-
+-import org.jboss.resteasy.annotations.jaxrs.PathParam;
+-
+ import io.smallrye.mutiny.Uni;
+ import io.vertx.core.Vertx;
 +import jakarta.inject.Inject;
 +import jakarta.ws.rs.GET;
 +import jakarta.ws.rs.Path;
++import jakarta.ws.rs.PathParam;
 +import jakarta.ws.rs.Produces;
 +import jakarta.ws.rs.core.MediaType;
++
++import static java.util.concurrent.TimeUnit.MILLISECONDS;
++import static java.util.concurrent.TimeUnit.NANOSECONDS;
  
- import org.jboss.resteasy.annotations.jaxrs.PathParam;
- 
+ @Path("/hello")
+ public class GreetingResource {
+@@ -38,7 +36,7 @@ public class GreetingResource {
+     @GET
+     @Produces(MediaType.TEXT_PLAIN)
+     @Path("{name}")
+-    public Uni<String> greeting(@PathParam String name) {
++    public Uni<String> greeting(@PathParam(value = "name") String name) {
+         return Uni.createFrom().emitter(emitter -> {
+             long start = System.nanoTime();
+             vertx.setTimer(10, l -> {
 diff --git a/apps/quarkus-vertx/src/main/java/org/acme/vertx/GreetingService.java b/apps/quarkus-vertx/src/main/java/org/acme/vertx/GreetingService.java
 index 0cb587a..ba16e14 100644
 --- a/apps/quarkus-vertx/src/main/java/org/acme/vertx/GreetingService.java
@@ -70,10 +182,10 @@ index 0cb587a..ba16e14 100644
  import io.quarkus.vertx.ConsumeEvent;
  
 diff --git a/apps/quarkus-vertx/src/main/java/org/acme/vertx/VertxJsonResource.java b/apps/quarkus-vertx/src/main/java/org/acme/vertx/VertxJsonResource.java
-index cc9c525..3650581 100644
+index cc9c525..4fc2fd4 100644
 --- a/apps/quarkus-vertx/src/main/java/org/acme/vertx/VertxJsonResource.java
 +++ b/apps/quarkus-vertx/src/main/java/org/acme/vertx/VertxJsonResource.java
-@@ -15,10 +15,10 @@
+@@ -15,13 +15,12 @@
   */
  package org.acme.vertx;
  
@@ -81,10 +193,94 @@ index cc9c525..3650581 100644
 -import javax.ws.rs.Path;
 -import javax.ws.rs.Produces;
 -import javax.ws.rs.core.MediaType;
+-
+-import org.jboss.resteasy.annotations.jaxrs.PathParam;
 +import jakarta.ws.rs.GET;
 +import jakarta.ws.rs.Path;
 +import jakarta.ws.rs.Produces;
 +import jakarta.ws.rs.core.MediaType;
  
- import org.jboss.resteasy.annotations.jaxrs.PathParam;
++import jakarta.ws.rs.PathParam;
+ import io.vertx.core.json.JsonArray;
+ import io.vertx.core.json.JsonObject;
  
+@@ -31,13 +30,13 @@ public class VertxJsonResource {
+ 
+     @GET
+     @Path("{name}/object")
+-    public JsonObject jsonObject(@PathParam String name) {
++    public JsonObject jsonObject(@PathParam(value = "name") String name) {
+         return new JsonObject().put("Hello", name);
+     }
+ 
+     @GET
+     @Path("{name}/array")
+-    public JsonArray jsonArray(@PathParam String name) {
++    public JsonArray jsonArray(@PathParam(value = "name") String name) {
+         return new JsonArray().add("Hello").add(name);
+     }
+ }
+diff --git a/apps/quarkus-vertx/src/main/resources/application.properties b/apps/quarkus-vertx/src/main/resources/application.properties
+index 33ee4a9..138bc8d 100644
+--- a/apps/quarkus-vertx/src/main/resources/application.properties
++++ b/apps/quarkus-vertx/src/main/resources/application.properties
+@@ -5,4 +5,6 @@ quarkus.datasource.username=quarkus_test
+ quarkus.datasource.password=quarkus_test
+ quarkus.datasource.reactive.url=postgresql://localhost:5432/quarkus_test
+ 
+-myapp.schema.create=true
+\ No newline at end of file
++myapp.schema.create=true
++# A very long time, we hang in there during debugging.
++quarkus.vertx.max-event-loop-execute-time=2M
+diff --git a/apps/quarkus-vertx/src/main/java/org/acme/vertx/FruitApp.java b/apps/quarkus-vertx/src/main/java/org/acme/vertx/FruitApp.java
+new file mode 100644
+index 0000000..b772382
+--- /dev/null
++++ b/apps/quarkus-vertx/src/main/java/org/acme/vertx/FruitApp.java
+@@ -0,0 +1,45 @@
++/*
++ * Copyright 2023 Red Hat, Inc.
++ *
++ * Red Hat licenses this file to you under the Apache License, version 2.0
++ * (the "License"); you may not use this file except in compliance with the
++ * License.  You may obtain a copy of the License at:
++ *
++ * http://www.apache.org/licenses/LICENSE-2.0
++ *
++ * Unless required by applicable law or agreed to in writing, software
++ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
++ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
++ * License for the specific language governing permissions and limitations
++ * under the License.
++ */
++
++package org.acme.vertx;
++
++import io.quarkus.runtime.Startup;
++import io.vertx.mutiny.pgclient.PgPool;
++import jakarta.enterprise.context.ApplicationScoped;
++import jakarta.inject.Inject;
++import org.eclipse.microprofile.config.inject.ConfigProperty;
++
++@ApplicationScoped
++public class FruitApp {
++
++    @Inject
++    @ConfigProperty(name = "myapp.schema.create", defaultValue = "true")
++    boolean schemaCreate;
++
++    @Inject
++    PgPool client;
++
++    @Startup
++    void initdb() {
++        if (schemaCreate) {
++            client.query("DROP TABLE IF EXISTS fruits").execute()
++                    .flatMap(r -> client.query("CREATE TABLE fruits (id SERIAL PRIMARY KEY, name TEXT NOT NULL)").execute())
++                    .flatMap(r -> client.query("INSERT INTO fruits (name) VALUES ('Orange')").execute())
++                    .flatMap(r -> client.query("INSERT INTO fruits (name) VALUES ('Pear')").execute())
++                    .flatMap(r -> client.query("INSERT INTO fruits (name) VALUES ('Apple')").execute()).await().indefinitely();
++        }
++    }
++}

--- a/pom.xml
+++ b/pom.xml
@@ -29,7 +29,7 @@
         <!-- Quarkus apps -->
         <!-- Note that the TS switches Quarkus version at test run,
              see Commands.java, QUARKUS_VERSION -->
-        <global.quarkus.version>2.13.7.Final</global.quarkus.version>
+        <global.quarkus.version>2.13.9.Final</global.quarkus.version>
 
         <!-- Helidon quickstart -->
         <version.helidon>2.2.2</version.helidon>

--- a/testsuite/src/it/java/org/graalvm/tests/integration/AppReproducersTest.java
+++ b/testsuite/src/it/java/org/graalvm/tests/integration/AppReproducersTest.java
@@ -908,27 +908,9 @@ public class AppReproducersTest {
             // Build
             processLog = Path.of(appDir.getAbsolutePath(), "logs", "build-and-run.log").toFile();
 
-            Map<String, String> switches = new HashMap<>();
-            Version version = UsedVersion.getVersion(app.runtimeContainer != ContainerNames.NONE);
-            if (version.compareTo(Version.create(23, 1, 0)) >= 0) {
-                switches.put(UnlockExperimentalVMOptions_23_1.token, UnlockExperimentalVMOptions_23_1.replacement);
-                switches.put(LockExperimentalVMOptions_23_1.token, LockExperimentalVMOptions_23_1.replacement);
-            } else {
-                switches.put(UnlockExperimentalVMOptions_23_1.token, "");
-                switches.put(LockExperimentalVMOptions_23_1.token, "");
-            }
-            if (version.compareTo(Version.create(23, 0, 0)) >= 0) {
-                switches.put(TrackNodeSourcePosition_23_0.token, TrackNodeSourcePosition_23_0.replacement);
-                switches.put(DebugCodeInfoUseSourceMappings_23_0.token, DebugCodeInfoUseSourceMappings_23_0.replacement);
-                switches.put(OmitInlinedMethodDebugLineInfo_23_0.token, OmitInlinedMethodDebugLineInfo_23_0.replacement);
-            } else {
-                switches.put(TrackNodeSourcePosition_23_0.token, "");
-                switches.put(DebugCodeInfoUseSourceMappings_23_0.token, "");
-                switches.put(OmitInlinedMethodDebugLineInfo_23_0.token, "");
-            }
             // In this case, the two last commands are used for running the app; one in JVM mode and the other in Native mode.
             // We should somehow capture this semantically in an Enum or something. This is fragile...
-            builderRoutine(app.buildAndRunCmds.cmds.length - 2, app, report, cn, mn, appDir, processLog, null, switches);
+            builderRoutine(app.buildAndRunCmds.cmds.length - 2, app, report, cn, mn, appDir, processLog, null, getSwitches(app));
 
             final File inputData = new File(BASE_DIR + File.separator + app.dir + File.separator + "target" + File.separator + "test_data.txt");
 
@@ -965,8 +947,8 @@ public class AppReproducersTest {
             }
 
             assertEquals(2, count, "There were two same hashes " + magicHash + " expected in the log. " +
-                    "One from JVM run and one for Native image run. " +
-                    "" + count + " such hashes were found. Check build-and-run.log and report.md.");
+                    "One from JVM run and one for Native image run. " + count +
+                    " such hashes were found. Check build-and-run.log and report.md.");
 
             processStopper(process, false);
             Logs.checkLog(cn, mn, app, processLog);
@@ -991,5 +973,27 @@ public class AppReproducersTest {
         } finally {
             cleanup(process, cn, mn, report, app, processLog);
         }
+    }
+
+    private static Map<String, String> getSwitches(Apps app) {
+        final Map<String, String> switches = new HashMap<>();
+        final Version version = UsedVersion.getVersion(app.runtimeContainer != ContainerNames.NONE);
+        if (version.compareTo(Version.create(23, 1, 0)) >= 0) {
+            switches.put(UnlockExperimentalVMOptions_23_1.token, UnlockExperimentalVMOptions_23_1.replacement);
+            switches.put(LockExperimentalVMOptions_23_1.token, LockExperimentalVMOptions_23_1.replacement);
+        } else {
+            switches.put(UnlockExperimentalVMOptions_23_1.token, "");
+            switches.put(LockExperimentalVMOptions_23_1.token, "");
+        }
+        if (version.compareTo(Version.create(23, 0, 0)) >= 0) {
+            switches.put(TrackNodeSourcePosition_23_0.token, TrackNodeSourcePosition_23_0.replacement);
+            switches.put(DebugCodeInfoUseSourceMappings_23_0.token, DebugCodeInfoUseSourceMappings_23_0.replacement);
+            switches.put(OmitInlinedMethodDebugLineInfo_23_0.token, OmitInlinedMethodDebugLineInfo_23_0.replacement);
+        } else {
+            switches.put(TrackNodeSourcePosition_23_0.token, "");
+            switches.put(DebugCodeInfoUseSourceMappings_23_0.token, "");
+            switches.put(OmitInlinedMethodDebugLineInfo_23_0.token, "");
+        }
+        return switches;
     }
 }

--- a/testsuite/src/it/java/org/graalvm/tests/integration/DebugSymbolsTest.java
+++ b/testsuite/src/it/java/org/graalvm/tests/integration/DebugSymbolsTest.java
@@ -43,7 +43,10 @@ import java.io.OutputStreamWriter;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.util.Date;
 import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
 import java.util.concurrent.ConcurrentLinkedQueue;
 import java.util.concurrent.ExecutorService;
@@ -131,27 +134,9 @@ public class DebugSymbolsTest {
             // Build
             processLog = Path.of(appDir.getAbsolutePath(), "logs", "build-and-run.log").toFile();
 
-            Map<String, String> switches = new HashMap<>();
-            Version version = UsedVersion.getVersion(false);
-            if (version.compareTo(Version.create(23, 1, 0)) >= 0) {
-                switches.put(UnlockExperimentalVMOptions_23_1.token, UnlockExperimentalVMOptions_23_1.replacement);
-                switches.put(LockExperimentalVMOptions_23_1.token, LockExperimentalVMOptions_23_1.replacement);
-            } else {
-                switches.put(UnlockExperimentalVMOptions_23_1.token, "");
-                switches.put(LockExperimentalVMOptions_23_1.token, "");
-            }
-            if (version.compareTo(Version.create(23, 0, 0)) >= 0) {
-                switches.put(TrackNodeSourcePosition_23_0.token, TrackNodeSourcePosition_23_0.replacement);
-                switches.put(DebugCodeInfoUseSourceMappings_23_0.token, DebugCodeInfoUseSourceMappings_23_0.replacement);
-                switches.put(OmitInlinedMethodDebugLineInfo_23_0.token, OmitInlinedMethodDebugLineInfo_23_0.replacement);
-            } else {
-                switches.put(TrackNodeSourcePosition_23_0.token, "");
-                switches.put(DebugCodeInfoUseSourceMappings_23_0.token, "");
-                switches.put(OmitInlinedMethodDebugLineInfo_23_0.token, "");
-            }
             // In this case, the two last commands are used for running the app; one in JVM mode and the other in Native mode.
             // We should somehow capture this semantically in an Enum or something. This is fragile...
-            builderRoutine(app.buildAndRunCmds.cmds.length - 2, app, report, cn, mn, appDir, processLog, null, switches);
+            builderRoutine(app.buildAndRunCmds.cmds.length - 2, app, report, cn, mn, appDir, processLog, null, getSwitches());
 
             assertTrue(Files.exists(Path.of(appDir.getAbsolutePath(), "target", "debug-symbols-smoke")),
                     "debug-symbols-smoke executable does not exist. Compilation failed. Check the logs.");
@@ -201,6 +186,28 @@ public class DebugSymbolsTest {
         } finally {
             cleanup(null, cn, mn, report, app, processLog);
         }
+    }
+
+    private static Map<String, String> getSwitches() {
+        final Map<String, String> switches = new HashMap<>();
+        final Version version = UsedVersion.getVersion(false);
+        if (version.compareTo(Version.create(23, 1, 0)) >= 0) {
+            switches.put(UnlockExperimentalVMOptions_23_1.token, UnlockExperimentalVMOptions_23_1.replacement);
+            switches.put(LockExperimentalVMOptions_23_1.token, LockExperimentalVMOptions_23_1.replacement);
+        } else {
+            switches.put(UnlockExperimentalVMOptions_23_1.token, "");
+            switches.put(LockExperimentalVMOptions_23_1.token, "");
+        }
+        if (version.compareTo(Version.create(23, 0, 0)) >= 0) {
+            switches.put(TrackNodeSourcePosition_23_0.token, TrackNodeSourcePosition_23_0.replacement);
+            switches.put(DebugCodeInfoUseSourceMappings_23_0.token, DebugCodeInfoUseSourceMappings_23_0.replacement);
+            switches.put(OmitInlinedMethodDebugLineInfo_23_0.token, OmitInlinedMethodDebugLineInfo_23_0.replacement);
+        } else {
+            switches.put(TrackNodeSourcePosition_23_0.token, "");
+            switches.put(DebugCodeInfoUseSourceMappings_23_0.token, "");
+            switches.put(OmitInlinedMethodDebugLineInfo_23_0.token, "");
+        }
+        return switches;
     }
 
     @Test
@@ -327,10 +334,12 @@ public class DebugSymbolsTest {
         final String cn = testInfo.getTestClass().get().getCanonicalName();
         final String mn = testInfo.getTestMethod().get().getName();
         final Pattern dbReady = Pattern.compile(".*ready to accept connections.*");
+        final Pattern appStarted = Pattern.compile(".*started.*");
         try {
             // Cleanup
             cleanTarget(app);
             stopAllRunningContainers();
+            removeContainers(app.runtimeContainer.name, "quarkus_test_db");
             Files.createDirectories(Paths.get(appDir.getAbsolutePath() + File.separator + "logs"));
 
             if (applySourcesPatch()) {
@@ -340,14 +349,30 @@ public class DebugSymbolsTest {
                 runCommand(getRunCommand("git", "apply", "quarkus_3.x.patch"), appDir);
             }
 
-            // Build & Run
+            // Build app and start db
             processLog = Path.of(appDir.getAbsolutePath(), "logs", "build-and-run.log").toFile();
-            builderRoutine(app.buildAndRunCmds.cmds.length, app, report, cn, mn, appDir, processLog);
-
+            builderRoutine(app.buildAndRunCmds.cmds.length - 1, app, report, cn, mn, appDir, processLog);
             waitForContainerLogToMatch("quarkus_test_db", dbReady, 20, 1, TimeUnit.SECONDS);
+
+            // Start app
+            LOGGER.info("Running...");
+            final List<String> cmd = getRunCommand(app.buildAndRunCmds.cmds[app.buildAndRunCmds.cmds.length - 1]);
+            runCommand(cmd, appDir, processLog, app);
+            Files.writeString(processLog.toPath(), String.join(" ", cmd) + "\n", StandardOpenOption.APPEND,
+                    StandardOpenOption.CREATE);
+            Logs.appendln(report, (new Date()).toString());
+            Logs.appendln(report, appDir.getAbsolutePath());
+            Logs.appendlnSection(report, String.join(" ", cmd));
+            waitForContainerLogToMatch(app.runtimeContainer.name, appStarted, 5, 1, TimeUnit.SECONDS);
+
+            // Sanity check that the app works before we commence debugging
+            LOGGER.info("Testing web page content...");
+            WebpageTester.testWeb(app.urlContent.urlContent[0][0], 20, app.urlContent.urlContent[0][1], false);
 
             // Check the log now to make sure there are no install failures
             // before gdb session starts.
+            runCommand(getRunCommand(CONTAINER_RUNTIME, "logs", app.runtimeContainer.name),
+                    appDir, processLog, app).waitFor(5, TimeUnit.SECONDS);
             Logs.checkLog(cn, mn, app, processLog);
 
             // GDB process
@@ -355,7 +380,8 @@ public class DebugSymbolsTest {
             // Q 1.11.7.Final and Mandrel 20.3.2 needs work/application.debug
             // Is Q 2.x baking debug symbols to the main executable too?
             final ProcessBuilder processBuilder = new ProcessBuilder(getRunCommand(
-                    CONTAINER_RUNTIME, "exec", "-i", ContainerNames.QUARKUS_BUILDER_IMAGE_ENCODING.name, "/usr/bin/gdb", "--interpreter=mi", "/work/application.debug", "1"))
+                    CONTAINER_RUNTIME, "exec", "-i", ContainerNames.QUARKUS_BUILDER_IMAGE_ENCODING.name, "/usr/bin/gdb",
+                    "--interpreter=mi", "/work/application.debug", "1"))
                     .directory(appDir)
                     .redirectErrorStream(true);
             final Map<String, String> envA = processBuilder.environment();
@@ -400,10 +426,8 @@ public class DebugSymbolsTest {
 
             gdbProcess.waitFor(1, TimeUnit.SECONDS);
 
-            final Process process = runCommand(
-                    getRunCommand(CONTAINER_RUNTIME, "logs", app.runtimeContainer.name),
-                    appDir, processLog, app);
-            process.waitFor(5, TimeUnit.SECONDS);
+            runCommand(getRunCommand(CONTAINER_RUNTIME, "logs", app.runtimeContainer.name),
+                    appDir, processLog, app).waitFor(5, TimeUnit.SECONDS);
 
             processStopper(gdbProcess, true);
             stopRunningContainers(app.runtimeContainer.name, "quarkus_test_db");

--- a/testsuite/src/it/java/org/graalvm/tests/integration/PerfCheckTest.java
+++ b/testsuite/src/it/java/org/graalvm/tests/integration/PerfCheckTest.java
@@ -144,19 +144,7 @@ public class PerfCheckTest {
             }
 
             // Build executables
-            final Map<String, String> switches;
-            if (UsedVersion.getVersion(false).compareTo(Version.create(22, 2, 0)) >= 0) {
-                switches = Map.of(
-                        GRAALVM_BUILD_OUTPUT_JSON_FILE + "-ParseOnce", "," + GRAALVM_BUILD_OUTPUT_JSON_FILE_SWITCH + "quarkus-json_minus-ParseOnce.json",
-                        GRAALVM_BUILD_OUTPUT_JSON_FILE + "+ParseOnce", "," + GRAALVM_BUILD_OUTPUT_JSON_FILE_SWITCH + "quarkus-json_plus-ParseOnce.json"
-                );
-            } else {
-                switches = Map.of(
-                        GRAALVM_BUILD_OUTPUT_JSON_FILE + "-ParseOnce", "",
-                        GRAALVM_BUILD_OUTPUT_JSON_FILE + "+ParseOnce", ""
-                );
-            }
-            builderRoutine(3, app, null, null, null, appDir, processLog, null, switches);
+            builderRoutine(3, app, null, null, null, appDir, processLog, null, getSwitches1());
             assertTrue(processLog.exists());
 
             int line = 0;
@@ -287,13 +275,7 @@ public class PerfCheckTest {
             }
 
             // Build executables
-            final Map<String, String> switches;
-            if (UsedVersion.getVersion(false).compareTo(Version.create(22, 2, 0)) >= 0) {
-                switches = Map.of(GRAALVM_BUILD_OUTPUT_JSON_FILE, "," + GRAALVM_BUILD_OUTPUT_JSON_FILE_SWITCH + "quarkus-json.json");
-            } else {
-                switches = Map.of(GRAALVM_BUILD_OUTPUT_JSON_FILE, "");
-            }
-            builderRoutine(2, app, null, null, null, appDir, processLog, null, switches);
+            builderRoutine(2, app, null, null, null, appDir, processLog, null, getSwitches2());
 
             int line = 0;
             for (int i = 2; i >= 1; i--) {
@@ -426,18 +408,7 @@ public class PerfCheckTest {
             }
 
             // Build executables
-            final Map<String, String> switches = new HashMap<>();
-            if (UsedVersion.getVersion(false).compareTo(Version.create(22, 2, 0)) >= 0) {
-                if (UsedVersion.getVersion(false).compareTo(Version.create(23, 1, 0)) >= 0) {
-                    switches.put(GRAALVM_BUILD_OUTPUT_JSON_FILE, ",-H:+UnlockExperimentalVMOptions," + GRAALVM_BUILD_OUTPUT_JSON_FILE_SWITCH + "quarkus-json.json,-H:-UnlockExperimentalVMOptions");
-                    switches.put("-H:Log=registerResource:", "-H:+UnlockExperimentalVMOptions,-H:Log=registerResource:,-H:-UnlockExperimentalVMOptions");
-                } else {
-                    switches.put(GRAALVM_BUILD_OUTPUT_JSON_FILE, "," + GRAALVM_BUILD_OUTPUT_JSON_FILE_SWITCH + "quarkus-json.json");
-                }
-            } else {
-                switches.put(GRAALVM_BUILD_OUTPUT_JSON_FILE, "");
-            }
-            builderRoutine(2, app, null, null, null, appDir, processLog, null, switches);
+            builderRoutine(2, app, null, null, null, appDir, processLog, null, getSwitches3());
 
             int line = 0;
             for (int i = 2; i >= 1; i--) {
@@ -528,5 +499,67 @@ public class PerfCheckTest {
                 runCommand(getRunCommand("git", "apply", "-R", patch), appDir);
             }
         }
+    }
+
+    private static Map<String, String> getSwitches1() {
+        final Map<String, String> switches;
+        if (UsedVersion.getVersion(false).compareTo(Version.create(22, 2, 0)) >= 0) {
+            if (UsedVersion.getVersion(false).compareTo(Version.create(23, 1, 0)) >= 0) {
+                switches = Map.of(
+                        GRAALVM_BUILD_OUTPUT_JSON_FILE + "-ParseOnce",
+                        ",-H:+UnlockExperimentalVMOptions," + GRAALVM_BUILD_OUTPUT_JSON_FILE_SWITCH + "quarkus-json_minus-ParseOnce.json,-H:-UnlockExperimentalVMOptions",
+                        GRAALVM_BUILD_OUTPUT_JSON_FILE + "+ParseOnce",
+                        ",-H:+UnlockExperimentalVMOptions," + GRAALVM_BUILD_OUTPUT_JSON_FILE_SWITCH + "quarkus-json_plus-ParseOnce.json,-H:-UnlockExperimentalVMOptions",
+                        "-H:-ParseOnce", "-H:+UnlockExperimentalVMOptions,-H:-ParseOnce,-H:-UnlockExperimentalVMOptions",
+                        "-H:+ParseOnce", "-H:+UnlockExperimentalVMOptions,-H:+ParseOnce,-H:-UnlockExperimentalVMOptions"
+                );
+            } else {
+                switches = Map.of(
+                        GRAALVM_BUILD_OUTPUT_JSON_FILE + "-ParseOnce",
+                        "," + GRAALVM_BUILD_OUTPUT_JSON_FILE_SWITCH + "quarkus-json_minus-ParseOnce.json",
+                        GRAALVM_BUILD_OUTPUT_JSON_FILE + "+ParseOnce",
+                        "," + GRAALVM_BUILD_OUTPUT_JSON_FILE_SWITCH + "quarkus-json_plus-ParseOnce.json"
+                );
+            }
+        } else {
+            switches = Map.of(
+                    GRAALVM_BUILD_OUTPUT_JSON_FILE + "-ParseOnce", "",
+                    GRAALVM_BUILD_OUTPUT_JSON_FILE + "+ParseOnce", ""
+            );
+        }
+        return switches;
+    }
+
+    private static Map<String, String> getSwitches2() {
+        final Map<String, String> switches;
+        if (UsedVersion.getVersion(false).compareTo(Version.create(22, 2, 0)) >= 0) {
+            if (UsedVersion.getVersion(false).compareTo(Version.create(23, 1, 0)) >= 0) {
+                switches = Map.of(GRAALVM_BUILD_OUTPUT_JSON_FILE,
+                        ",-H:+UnlockExperimentalVMOptions," + GRAALVM_BUILD_OUTPUT_JSON_FILE_SWITCH + "quarkus-json.json,-H:-UnlockExperimentalVMOptions");
+            } else {
+                switches = Map.of(GRAALVM_BUILD_OUTPUT_JSON_FILE,
+                        "," + GRAALVM_BUILD_OUTPUT_JSON_FILE_SWITCH + "quarkus-json.json");
+            }
+        } else {
+            switches = Map.of(GRAALVM_BUILD_OUTPUT_JSON_FILE, "");
+        }
+        return switches;
+    }
+
+    private static Map<String, String> getSwitches3() {
+        final Map<String, String> switches = new HashMap<>();
+        if (UsedVersion.getVersion(false).compareTo(Version.create(22, 2, 0)) >= 0) {
+            if (UsedVersion.getVersion(false).compareTo(Version.create(23, 1, 0)) >= 0) {
+                switches.put(GRAALVM_BUILD_OUTPUT_JSON_FILE,
+                        ",-H:+UnlockExperimentalVMOptions," + GRAALVM_BUILD_OUTPUT_JSON_FILE_SWITCH + "quarkus-json.json,-H:-UnlockExperimentalVMOptions");
+                switches.put("-H:Log=registerResource:",
+                        "-H:+UnlockExperimentalVMOptions,-H:Log=registerResource:,-H:-UnlockExperimentalVMOptions");
+            } else {
+                switches.put(GRAALVM_BUILD_OUTPUT_JSON_FILE, "," + GRAALVM_BUILD_OUTPUT_JSON_FILE_SWITCH + "quarkus-json.json");
+            }
+        } else {
+            switches.put(GRAALVM_BUILD_OUTPUT_JSON_FILE, "");
+        }
+        return switches;
     }
 }

--- a/testsuite/src/it/java/org/graalvm/tests/integration/RuntimesSmokeTest.java
+++ b/testsuite/src/it/java/org/graalvm/tests/integration/RuntimesSmokeTest.java
@@ -101,7 +101,7 @@ public class RuntimesSmokeTest {
 
             // Run
             LOGGER.info("Running...");
-            List<String> cmd = getRunCommand(app.buildAndRunCmds.cmds[app.buildAndRunCmds.cmds.length - 1]);
+            final List<String> cmd = getRunCommand(app.buildAndRunCmds.cmds[app.buildAndRunCmds.cmds.length - 1]);
             process = runCommand(cmd, appDir, processLog, app);
             Logs.appendln(report, appDir.getAbsolutePath());
             Logs.appendlnSection(report, String.join(" ", cmd));

--- a/testsuite/src/it/java/org/graalvm/tests/integration/utils/Apps.java
+++ b/testsuite/src/it/java/org/graalvm/tests/integration/utils/Apps.java
@@ -70,7 +70,7 @@ public enum Apps {
             BuildAndRunCmds.DEBUG_QUARKUS_FULL_MICROPROFILE,
             ContainerNames.NONE),
     DEBUG_QUARKUS_BUILDER_IMAGE_VERTX("apps" + File.separator + "quarkus-vertx",
-            URLContent.NONE,
+            URLContent.DEBUG_QUARKUS_BUILDER_IMAGE_VERTX,
             WhitelistLogLines.DEBUG_QUARKUS_BUILDER_IMAGE_VERTX,
             BuildAndRunCmds.DEBUG_QUARKUS_BUILDER_IMAGE_VERTX,
             ContainerNames.DEBUG_QUARKUS_BUILDER_IMAGE_VERTX),

--- a/testsuite/src/it/java/org/graalvm/tests/integration/utils/BuildAndRunCmds.java
+++ b/testsuite/src/it/java/org/graalvm/tests/integration/utils/BuildAndRunCmds.java
@@ -129,11 +129,11 @@ public enum BuildAndRunCmds {
                     "--name", ContainerNames.QUARKUS_BUILDER_IMAGE_ENCODING.name, "my-quarkus-mandrel-app"}
     }),
     DEBUG_QUARKUS_BUILDER_IMAGE_VERTX(new String[][]{
+            new String[]{"mvn", "dependency:sources", "-Dquarkus.version=" + QUARKUS_VERSION.getVersionString()},
             new String[]{"mvn", "package", "-Pnative", "-Dquarkus.native.container-build=true",
                     "-Dquarkus.native.container-runtime=" + CONTAINER_RUNTIME,
                     "-Dquarkus.native.builder-image=" + BUILDER_IMAGE,
                     "-Dquarkus.native.debug.enabled=true", "-Dquarkus.version=" + QUARKUS_VERSION.getVersionString()},
-            new String[]{"mvn", "dependency:sources", "-Dquarkus.version=" + QUARKUS_VERSION.getVersionString()},
             new String[]{CONTAINER_RUNTIME, "build", "--network=host", "-f", "src/main/docker/Dockerfile.native", "-t", "my-quarkus-mandrel-app", "."},
             new String[]{CONTAINER_RUNTIME, "run", "--network=host", "--ulimit", "memlock=-1:-1", "-it", "-d", "--rm=true",
                     "--name", "quarkus_test_db", "-e", "POSTGRES_USER=quarkus_test", "-e", "POSTGRES_PASSWORD=quarkus_test",

--- a/testsuite/src/it/java/org/graalvm/tests/integration/utils/Commands.java
+++ b/testsuite/src/it/java/org/graalvm/tests/integration/utils/Commands.java
@@ -357,9 +357,9 @@ public class Commands {
     }
 
     public static boolean waitForContainerLogToMatch(String containerName, Pattern pattern, long timeout, long sleep, TimeUnit unit) throws IOException, InterruptedException {
-        long timeoutMillis = unit.toMillis(timeout);
-        long sleepMillis = unit.toMillis(sleep);
-        long startMillis = System.currentTimeMillis();
+        final long timeoutMillis = unit.toMillis(timeout);
+        final long sleepMillis = unit.toMillis(sleep);
+        final long startMillis = System.currentTimeMillis();
         final List<String> cmd = getRunCommand(CONTAINER_RUNTIME, "logs", containerName);
         LOGGER.infof("Command: %s", cmd);
         final ProcessBuilder processBuilder = new ProcessBuilder(cmd);
@@ -367,7 +367,7 @@ public class Commands {
         envA.put("PATH", System.getenv("PATH"));
         processBuilder.redirectErrorStream(true);
         while (System.currentTimeMillis() - startMillis < timeoutMillis) {
-            Process p = processBuilder.start();
+            final Process p = processBuilder.start();
             try (BufferedReader processOutputReader =
                          new BufferedReader(new InputStreamReader(p.getInputStream(), StandardCharsets.UTF_8))) {
                 String l;

--- a/testsuite/src/it/java/org/graalvm/tests/integration/utils/GDBSession.java
+++ b/testsuite/src/it/java/org/graalvm/tests/integration/utils/GDBSession.java
@@ -260,11 +260,15 @@ public enum GDBSession {
         @Override
         public CP[] get(boolean inContainer) {
             // The huge timeout is needed because it takes a very long time to set a breakpoint, even after: https://github.com/graalvm/mandrel/pull/545
-            final long increasedTimeoutMs = (UsedVersion.getVersion(inContainer).compareTo(Version.create(23, 0, 0)) >= 0) ? CMD_LONG_TIMEOUT_MS : CMD_DEFAULT_TIMEOUT_MS;
-            return new CP[]{
+            final long increasedTimeoutMs = (UsedVersion.getVersion(inContainer).compareTo(Version.create(23, 0, 0)) >= 0)
+                    ? CMD_LONG_TIMEOUT_MS
+                    : CMD_DEFAULT_TIMEOUT_MS;
+            return new CP[] {
                     SHOW_VERSION,
                     new CP("b Fruit.java:48\n",
-                            Pattern.compile(".*Breakpoint 1.*file org/acme/vertx/Fruit.java,.*line 48.*", Pattern.DOTALL), increasedTimeoutMs),
+                            Pattern.compile(".*Breakpoint 1.*file[ =\"]*org/acme/vertx/Fruit.java\"?,.*line[ =\"]*48.*",
+                                    Pattern.DOTALL),
+                            increasedTimeoutMs),
                     new CP("c&\n",
                             Pattern.compile(".*", Pattern.DOTALL)),
                     new CP("GOTO URL http://localhost:8080/fruits",
@@ -272,7 +276,8 @@ public enum GDBSession {
                     new CP("bt\n",
                             Pattern.compile(".*at org/acme/vertx/Fruit.java:48.*", Pattern.DOTALL)),
                     new CP("list\n",
-                            Pattern.compile(".*48.*return client.query\\(\"SELECT id, name FROM fruits ORDER BY name ASC.*", Pattern.DOTALL)),
+                            Pattern.compile(".*48.*return client.query\\(\"SELECT id, name FROM fruits ORDER BY name ASC.*",
+                                    Pattern.DOTALL)),
                     new CP("c&\n",
                             Pattern.compile(".*Continuing.*", Pattern.DOTALL)),
             };

--- a/testsuite/src/it/java/org/graalvm/tests/integration/utils/URLContent.java
+++ b/testsuite/src/it/java/org/graalvm/tests/integration/utils/URLContent.java
@@ -60,12 +60,16 @@ public enum URLContent {
             new String[]{"http://localhost:8090/openapi", "openapi: 3.0.0"},
             new String[]{"http://localhost:8090/benchmark/jfr-hyperfoil/start"},
     }),
+
     HELIDON_QUICKSTART_SE(new String[][]{
             new String[]{"http://localhost:8080/greet", "Hello World!"},
             new String[]{"http://localhost:8080/greet/Karm", "Hello Karm!"},
             new String[]{"http://localhost:8080/health", "\"UP\""},
             new String[]{"http://localhost:8080/metrics", "availableProcessors"}
-    });
+    }),
+
+    DEBUG_QUARKUS_BUILDER_IMAGE_VERTX(new String[][]{
+        new String[]{"http://localhost:8080/fruits", "\"name\":\"Apple\""}});
 
     public final String[][] urlContent;
 

--- a/testsuite/src/it/java/org/graalvm/tests/integration/utils/WhitelistLogLines.java
+++ b/testsuite/src/it/java/org/graalvm/tests/integration/utils/WhitelistLogLines.java
@@ -48,7 +48,9 @@ public enum WhitelistLogLines {
                     Pattern.compile("WARNING: Please consider reporting this to the maintainers of com.google.inject.internal.cglib.*"),
                     Pattern.compile("WARNING: Please consider reporting this to the maintainers of com.fasterxml.jackson.databind.util.*"),
                     // JAVA_HOME (e.g. 17) != GRAALVM_HOME (e.g. 21)
-                    Pattern.compile(".*system modules path not set in conjunction with -source .*")
+                    Pattern.compile(".*system modules path not set in conjunction with -source .*"),
+                    // It's accompanied by a list of all warnings, so this particular one can be white-listed globally
+                    Pattern.compile(".*Warning: Please re-evaluate whether any experimental option is required, and either remove or unlock it\\..*"),
             };
         }
     },
@@ -119,7 +121,6 @@ public enum WhitelistLogLines {
                     Pattern.compile(".*Warning: Could not register io.netty.* queryAllPublicMethods for reflection.*"),
                     // netty 4 which doesn't have the relevant native config in the lib. See https://github.com/netty/netty/pull/13596
                     Pattern.compile(".*Warning: The option '-H:ReflectionConfigurationResources=META-INF/native-image/io\\.netty/netty-transport/reflection-config\\.json' is experimental and must be enabled via.*"),
-                    Pattern.compile(".*Warning: Please re-evaluate whether any experimental option is required, and either remove or unlock it\\..*"),
                     // We don't run any OpenTracing collector point for simplicity, hence the exception. Q 3.6.0+ specific.
                     Pattern.compile(".*Failed to export spans. The request could not be executed. Full error message: Connection refused:.*")
             };
@@ -149,6 +150,9 @@ public enum WhitelistLogLines {
                     Pattern.compile(".*Warning: Could not register io.netty.* queryAllPublicMethods for reflection.*"),
                     // https://github.com/quarkusio/quarkus/blob/2.13.7.Final/core/deployment/src/main/java/io/quarkus/deployment/OutputFilter.java#L27
                     Pattern.compile(".*io.quarkus.deployment.OutputFilter.*Stream is closed, ignoring and trying to continue.*"),
+                    // Deprecated/to be updated with Rest Easy Reactive
+                    Pattern.compile(".*The option '-H:ReflectionConfigurationResources=META-INF/native-image/io.netty/netty-transport/reflection-config.json' is experimental.*"),
+                    Pattern.compile(".*The option '-H:IncludeResourceBundles=yasson-messages' is experimental.*"),
             };
         }
     },
@@ -158,7 +162,6 @@ public enum WhitelistLogLines {
             return new Pattern[]{
                     // Experimental options not being unlocked, produces warnings, yet it's driven by the helidon-maven-plugin
                     Pattern.compile(".*The option '.*' is experimental and must be enabled via '-H:\\+UnlockExperimentalVMOptions' in the future.*"),
-                    Pattern.compile(".*Please re-evaluate whether any experimental option is required, and either remove or unlock it.*"),
                     // Unused argument on new Graal
                     Pattern.compile(".*Ignoring server-mode native-image argument --no-server.*"),
                     // --allow-incomplete-classpath not available in new GraalVM https://github.com/Karm/mandrel-integration-tests/issues/76
@@ -183,6 +186,8 @@ public enum WhitelistLogLines {
                     Pattern.compile(".*Warning: Could not register io.netty.* queryAllPublicMethods for reflection.*"),
                     // https://github.com/quarkusio/quarkus/blob/2.13.7.Final/core/deployment/src/main/java/io/quarkus/deployment/OutputFilter.java#L27
                     Pattern.compile(".*io.quarkus.deployment.OutputFilter.*Stream is closed, ignoring and trying to continue.*"),
+                    // Perf test uses netty 4 which doesn't have the relevant native config in the lib. See https://github.com/netty/netty/pull/13596
+                    Pattern.compile(".*Warning: The option '-H:ReflectionConfigurationResources=META-INF/native-image/io\\.netty/netty-transport/reflection-config\\.json' is experimental and must be enabled via.*"),
             };
         }
     },
@@ -209,7 +214,6 @@ public enum WhitelistLogLines {
                         Pattern.compile(".*io.net.boo.ServerBootstrap.*Failed to register an accepted channel:.*"),
                         // Perf test uses netty 4 which doesn't have the relevant native config in the lib. See https://github.com/netty/netty/pull/13596
                         Pattern.compile(".*Warning: The option '-H:ReflectionConfigurationResources=META-INF/native-image/io\\.netty/netty-transport/reflection-config\\.json' is experimental and must be enabled via.*"),
-                        Pattern.compile(".*Warning: Please re-evaluate whether any experimental option is required, and either remove or unlock it\\..*")
                 };
             }
         }

--- a/testsuite/src/it/java/org/graalvm/tests/integration/utils/versions/QuarkusVersion.java
+++ b/testsuite/src/it/java/org/graalvm/tests/integration/utils/versions/QuarkusVersion.java
@@ -32,7 +32,7 @@ public class QuarkusVersion implements Comparable<QuarkusVersion> {
     private final String gitSHA;
 
     public QuarkusVersion() {
-        this(getProperty("QUARKUS_VERSION", "2.13.7.Final"));
+        this(getProperty("QUARKUS_VERSION", "2.13.9.Final"));
     }
 
     public QuarkusVersion(String version) {

